### PR TITLE
Fix(user-collection): Do not call other events (put/validate) when a user is changed from login event.

### PR DESCRIPTION
**Pull request by andreialecu**

### CODE REVIEW

File: lib/resources/user-collection.js [hunk 1]
```diff
@@ -164,6 +164,9 @@ UserCollection.prototype.handleLogin = function (ctx) {
         // create a new context and set the body to our user so that we can call save on the collection
         var newCtx = _.clone(ctx); 
         newCtx.body = userClone;
+        // skip events when calling uc.save, so that validate and put is not called from this
+        // internal call
+        newCtx.internalSkipEvents = true;
```
Comment (ericfong) [inline lib/resources/user-collection.js]: add _ as prefix?
_internalSkipEvents

File: lib/resources/collection/index.js [hunk 2]
```diff
@@ -700,7 +700,9 @@ Collection.prototype.execCommands = function (type, obj, commands) {
   return this;
 };
 
-Collection.prototype.shouldRunEvent = function(ev, ctx) {
+Collection.prototype.shouldRunEvent = function (ev, ctx) {
+  if (ctx && ctx.internalSkipEvents) return false;
```
Comment (ericfong) [inline lib/resources/collection/index.js]: Seems hard to maintain

**Comment by andreialecu** [comment]
@ericfong: updated

**Comment by ericfong** [comment]
landed as 81574fcbd523c0c708f54226ad6e47ad64804386
Thanks

By the way, please use a clean branch to do PR. Will be better for rebasing

**Pull request closed without merging**